src/class-main.php

```diff
@@ -77,17 +81,105 @@ public function register_services() : void {
       return;
     }
 
-    $classes = $this->get_service_classes();
+    $this->services = $this->get_service_classes_with_di();
 
-    $this->services = array_map(
-      [ $this, 'instantiate_service' ],
-      $classes
-    );
     array_walk(
       $this->services,
-      function( Service $service ) {
-        $service->register();
+      function( $class ) {
+        if ( ! $class instanceof Service ) {
+          return;
+        }
+
+        $class->register();
+      }
+    );
+  }
+
+  /**
+   * Return array of services with Dependency Injection parameters.
+   *
+   * @return array
+   *
+   * @since 0.7.0 Init
+   */
+  private function get_service_classes_with_di() : array {
+    $services = $this->get_service_classes_prepared_array();
+
+    $container = $this->get_di_container( $services );
+
+    return array_map(
+      function( $class ) use ( $container ) {
+        return $container->get( $class );
+      },
+      array_keys( $services )
+    );
+  }
+
+  /**
+   * Get services classes array ad prepare it.
+   * Key should be class name, and value empty array or DI items.
+   *
+   * @return array
+   *
+   * @since 0.7.0 Init.
+   */
+  private function get_service_classes_prepared_array() : array {
+    $output = [];
+    foreach ( $this->get_service_classes() as $key => $value ) {
+      if ( gettype( $value ) !== 'array' ) {
+        $output[ $value ] = [];
+      } else {
+        $output[ $key ] = $value;
       }
+    }
+
+    return $output;
+  }
+
+  /**
+   * Implement PHP-DI.
+   * Create Object used for PHP-DI to create Dependency Injection methods.
```

Comment:
```suggestion
   * Build and return a DI container.
   * Wire all the dependencies automatically, based on the provided array of class => dependencies from the get_di_items().
```

Something like this.